pkg/services/searchV2/service.go

```diff
@@ -175,5 +179,252 @@ func (s *StandardSearchService) DoDashboardQuery(ctx context.Context, user *back
 		return rsp
 	}
 
-	return doSearchQuery(ctx, s.logger, index, filter, q, s.extender.GetQueryExtender(q), s.cfg.AppSubURL)
+	response := doSearchQuery(ctx, s.logger, index, filter, q, s.extender.GetQueryExtender(q), s.cfg.AppSubURL)
+
+	if err := s.addAllowedActionsField(ctx, orgID, signedInUser, response); err != nil {
```

Comment:
can this be added optionally to queries rather than added to all results?

also since this has a lot of code related to actions... can we move that to its own file rather than having it within the service.go file?

Reply:
done both